Source/drlg_l1.cpp

```diff
@@ -2168,6 +2161,69 @@ void CryptPatternGroup7(int rndper)
 	PlaceMiniSetRandom(CryptPattern8, rndper);
 }
 
+bool PlaceCathedralStairs(lvl_entry entry)
+{
+	bool success = true;
+
+	// Place poison water entrance
+	if (Quests[Q_PWATER].IsAvailable()) {
+		if (!PlaceMiniSet(PWATERIN, 1, 1, 0, 0, entry == ENTRY_RTNLVL, -1))
+			success = false;
+		if (entry == ENTRY_RTNLVL)
+			ViewPosition.y--;

```

Comment:
The original actually sets the wrong position for quest level entrances. Instead `GetReturnLvlPos()` will overwrite it with the position where the player was when triggering the level translation.

We could get rid of one of them. Any thoughts?

Reply:
I have corrected it to give the correct position for PosionedWater, but I realize that it will also other adjustments for the other quests as most a positioned using .dun sections, except for laz portal.